firebase_admin/mlkit.py

```diff
@@ -459,6 +558,93 @@ def __init__(self, app):
         self._client = _http_client.JsonHttpClient(
             credential=app.credential.get_credential(),
             base_url=self._project_url)
+        self._operation_client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_MLKitService.OPERATION_URL)
+
+    def get_operation(self, op_name):
+        _validate_and_parse_operation_name(op_name)
+        try:
+            return self._operation_client.body('get', url=op_name)
+        except requests.exceptions.RequestException as error:
+            raise _utils.handle_platform_error_from_requests(error)
+
+    def _exponential_backoff(self, current_attempt, stop_time):
+        """Sleeps for the appropriate amount of time. Or thows deadline exceeded."""
+        delay_factor = pow(
+            _MLKitService.POLL_EXPONENTIAL_BACKOFF_FACTOR, current_attempt)
+        wait_time_seconds = delay_factor * _MLKitService.POLL_BASE_WAIT_TIME_SECONDS
+
+        if stop_time is not None:
+            max_seconds_left = (stop_time - datetime.datetime.now()).total_seconds()
+            if max_seconds_left < 1: # allow a bit of time for rpc
+                raise exceptions.DeadlineExceededError('Polling max time exceeded.')
+            else:
+                wait_time_seconds = min(wait_time_seconds, max_seconds_left - 1)
+        time.sleep(wait_time_seconds)
+
+
+    def handle_operation(self, operation, polling=False, max_time_seconds=None,
+                         always_return_model=True):
+        """Handles long running operations.
+
+        Args:
+            operation: The operation to handle.
+            polling: Should we allow polling for the operation to complete.
+            max_time_seconds: The maximum seconds to try polling for operation complete.
+                (None for no limit)
+            always_return_model: If true, returns a locked Model instead of raising deadline
+                exceeded exceptions.
+
+        Returns:
+            dict: A dictionary of the returned model properties.
+
+        Raises:
+            TypeError: if the operation is not a dictionary.
+            ValueError: If the operation is malformed.
+            err: If the operation exceeds polling attempts or stop_time
+        """
+        if not isinstance(operation, dict):
+            raise TypeError('Operation must be a dictionary.')
+        op_name = operation.get('name')
+        _, model_id = _validate_and_parse_operation_name(op_name)
+
+        current_attempt = 0
+        start_time = datetime.datetime.now()
+        stop_time = (None if max_time_seconds is None else
+                     start_time + datetime.timedelta(seconds=max_time_seconds))
+        while polling and not operation.get('done'):
```

Comment:
Nit: s/polling/wait_for_operation/ for clarity.

Reply:
done